diff --git a/relations/Cargo.toml b/relations/Cargo.toml
index 8536718..ab7d2ca 100644
--- a/relations/Cargo.toml
+++ b/relations/Cargo.toml
@@ -25,7 +25,7 @@ ark-secp256k1 = "0.4"
 ark-secq256k1 = "0.4"
 
 [features]
-default = ["asm", "parallel"]
+default = ["asm", "bench_prover" ]
 asm = ["ark-ff/asm", "bulletproofs/asm"]
 parallel = ["rayon", "ark-ec/parallel", "ark-ff/parallel", "ark-std/parallel", "bulletproofs/parallel"]
 bench_prover = []
@@ -36,26 +36,14 @@ table2 = ["usenix"]
 table3 = ["usenix"]
 all-tables = ["table1", "table2", "table3"]
 
-[[bench]]
-name = "batched_select_rerandomize"
-harness = false
-
-[[bench]]
-name = "select_rerandomize"
-harness = false
-
 [[bench]]
 name = "coin"
 harness = false
 
-[[bench]]
-name = "accumulator"
-harness = false
-
 [profile.release]
 lto = true
 
 [build]
  # Adds architecture specific optimizations including those enabling the asm backend for ark_ff. It could potentially cause slowdowns from other features.
 rustflags = "-C target-cpu=native"
- # You could alternatively add support for the asm backend through "-C target-feature=+bmi2,+adx", but the binary would only run on systems supporting these instructions.
\ No newline at end of file
+ # You could alternatively add support for the asm backend through "-C target-feature=+bmi2,+adx", but the binary would only run on systems supporting these instructions.
diff --git a/relations/benches/coin.rs b/relations/benches/coin.rs
index 49e276e..13c5b93 100644
--- a/relations/benches/coin.rs
+++ b/relations/benches/coin.rs
@@ -29,73 +29,19 @@ use ark_ec::short_weierstrass::Affine;
 use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
 use blake2::Blake2s256 as Blake2s;
 
-#[cfg(feature = "parallel")]
-use rayon::prelude::*;
-
-#[cfg(feature = "usenix")]
-fn bench_pour(c: &mut Criterion) {
-    #[cfg(feature = "table3")]
-    {
-        let threaded = "";
-
-        let curves = "pasta";
-        println!("Table 3\n");
-        println!("Benchmark Pour over the pasta cycle, |S|=2^20\n");
-        bench_pour_with_parameters::<1024, PallasBase, PallasConfig, VestaConfig>(
-            c, 2, 12, threaded, curves,
-        );
-        println!("Benchmark Pour over the pasta cycle, |S|=2^32\n");
-        bench_pour_with_parameters::<256, PallasBase, PallasConfig, VestaConfig>(
-            c, 4, 13, threaded, curves,
-        );
-        println!("Benchmark Pour over the pasta cycle, |S|=2^40\n");
-        bench_pour_with_parameters::<1024, PallasBase, PallasConfig, VestaConfig>(
-            c, 4, 13, threaded, curves,
-        );
-        let curves = "secp&q";
-        println!("Benchmark Pour over the secp256k1 / seqp256k1 cycle, |S|=2^20\n");
-        bench_pour_with_parameters::<1024, SecpBase, SecpConfig, SecqConfig>(
-            c, 2, 12, threaded, curves,
-        );
-        println!("Benchmark Pour over the secp256k1 / seqp256k1 cycle, |S|=2^32\n");
-        bench_pour_with_parameters::<256, SecpBase, SecpConfig, SecqConfig>(
-            c, 4, 13, threaded, curves,
-        );
-
-        println!("Benchmark Pour over the secp256k1 / seqp256k1 cycle, |S|=2^40\n");
-        bench_pour_with_parameters::<1024, SecpBase, SecpConfig, SecqConfig>(
-            c, 4, 13, threaded, curves,
-        );
-    }
-}
-
 #[cfg(not(feature = "usenix"))]
 fn bench_pour(c: &mut Criterion) {
     let threaded = {
-        #[cfg(feature = "parallel")]
-        {
-            "Multi_threaded"
-        }
         #[cfg(not(feature = "parallel"))]
         {
             "Single_threaded"
         }
     };
     let curves = "pasta";
-    bench_pour_with_parameters::<1024, PallasBase, PallasConfig, VestaConfig>(
-        c, 2, 12, threaded, curves,
-    );
-    bench_pour_with_parameters::<256, PallasBase, PallasConfig, VestaConfig>(
-        c, 4, 13, threaded, curves,
-    );
     bench_pour_with_parameters::<1024, PallasBase, PallasConfig, VestaConfig>(
         c, 4, 13, threaded, curves,
     );
     let curves = "secp&q";
-    bench_pour_with_parameters::<1024, SecpBase, SecpConfig, SecqConfig>(
-        c, 2, 12, threaded, curves,
-    );
-    bench_pour_with_parameters::<256, SecpBase, SecpConfig, SecqConfig>(c, 4, 13, threaded, curves);
     bench_pour_with_parameters::<1024, SecpBase, SecpConfig, SecqConfig>(
         c, 4, 13, threaded, curves,
     );
@@ -204,50 +150,6 @@ fn bench_pour_with_parameters<
 
         #[cfg(feature = "bench_prover")]
         group.bench_function("prove", |b| b.iter(|| prove()));
-
-        #[cfg(feature = "detailed_benchmarks")]
-        group.bench_function("verification_gadget", |b| {
-            b.iter(|| {
-                tx.clone().verification_gadget(
-                    b"select_and_rerandomize",
-                    &sr_params,
-                    &curve_tree,
-                    &schnorr_parameters,
-                );
-            })
-        });
-        #[cfg(feature = "detailed_benchmarks")]
-        group.bench_function("deserialize", |b| {
-            b.iter(|| {
-                let _pour =
-                    Pour::<L, P0, P1, Projective<P0>>::deserialize_compressed(&tx.pour_bytes[..])
-                        .unwrap();
-            })
-        });
-        #[cfg(feature = "detailed_benchmarks")]
-        group.bench_function("verify_single_with_deserialization", |b| {
-            b.iter(|| {
-                let (pallas_vt, vesta_vt) = tx.clone().verification_gadget(
-                    b"select_and_rerandomize",
-                    &sr_params,
-                    &curve_tree,
-                    &schnorr_parameters,
-                );
-
-                batch_verify(
-                    vec![pallas_vt],
-                    &sr_params.even_parameters.pc_gens,
-                    &sr_params.even_parameters.bp_gens,
-                )
-                .unwrap();
-                batch_verify(
-                    vec![vesta_vt],
-                    &sr_params.odd_parameters.pc_gens,
-                    &sr_params.odd_parameters.bp_gens,
-                )
-                .unwrap();
-            })
-        });
     }
 
     use std::iter;
@@ -259,89 +161,33 @@ fn bench_pour_with_parameters<
             &iter::repeat(pour_proof.clone()).take(n).collect::<Vec<_>>(),
             |b, proofs| {
                 b.iter(|| {
-                    #[cfg(not(feature = "parallel"))]
-                    {
-                        let mut pallas_verification_scalars_and_points =
-                            Vec::with_capacity(proofs.len());
-                        let mut vesta_verification_scalars_and_points =
-                            Vec::with_capacity(proofs.len());
-                        for proof in proofs {
-                            let p = proof.clone();
-                            let (pallas_vt, vesta_vt) = p.verification_gadget(
-                                b"select_and_rerandomize",
-                                &sr_params,
-                                &curve_tree,
-                            );
-
-                            pallas_verification_scalars_and_points.push(pallas_vt);
-                            vesta_verification_scalars_and_points.push(vesta_vt);
-                        }
-                        batch_verify(
-                            pallas_verification_scalars_and_points,
-                            &sr_params.even_parameters.pc_gens,
-                            &sr_params.even_parameters.bp_gens,
-                        )
-                        .unwrap();
-                        batch_verify(
-                            vesta_verification_scalars_and_points,
-                            &sr_params.odd_parameters.pc_gens,
-                            &sr_params.odd_parameters.bp_gens,
-                        )
-                        .unwrap();
-                    }
-                    #[cfg(feature = "parallel")]
-                    {
-                        let proofs_and_commitment_paths = proofs.par_iter().map(|proof| {
-                            let mut path_0 = proof.randomized_path_0.clone();
-                            let mut path_1 = proof.randomized_path_1.clone();
-                            curve_tree.select_and_rerandomize_verification_commitments(&mut path_0);
-                            curve_tree.select_and_rerandomize_verification_commitments(&mut path_1);
-                            (proof, path_0, path_1)
-                        });
-                        let proofs_and_commitment_paths_clone = proofs_and_commitment_paths.clone();
-                        rayon::join(
-                            || {
-                                // even verification tuples
-                                let event_vts: Vec<_> = proofs_and_commitment_paths
-                                    .map(|(proof, path0, path1)| {
-                                        proof.even_verification_gadget(
-                                            b"select_and_rerandomize",
-                                            &sr_params,
-                                            &path0,
-                                            &path1,
-                                            &curve_tree,
-                                        )
-                                    })
-                                    .collect();
-                                batch_verify(
-                                    event_vts,
-                                    &sr_params.even_parameters.pc_gens,
-                                    &sr_params.even_parameters.bp_gens,
-                                )
-                                .unwrap();
-                            },
-                            || {
-                                // odd verification tuples
-                                let odd_vts: Vec<_> = proofs_and_commitment_paths_clone
-                                    .map(|(proof, path0, path1)| {
-                                        proof.odd_verification_gadget(
-                                            b"select_and_rerandomize",
-                                            &sr_params,
-                                            &path0,
-                                            &path1,
-                                            &curve_tree,
-                                        )
-                                    })
-                                    .collect();
-                                batch_verify(
-                                    odd_vts,
-                                    &sr_params.odd_parameters.pc_gens,
-                                    &sr_params.odd_parameters.bp_gens,
-                                )
-                                .unwrap();
-                            },
-                        )
+                    let mut pallas_verification_scalars_and_points =
+                        Vec::with_capacity(proofs.len());
+                    let mut vesta_verification_scalars_and_points =
+                        Vec::with_capacity(proofs.len());
+                    for proof in proofs {
+                        let p = proof.clone();
+                        let (pallas_vt, vesta_vt) = p.verification_gadget(
+                            b"select_and_rerandomize",
+                            &sr_params,
+                            &curve_tree,
+                        );
+
+                        pallas_verification_scalars_and_points.push(pallas_vt);
+                        vesta_verification_scalars_and_points.push(vesta_vt);
                     }
+                    batch_verify(
+                        pallas_verification_scalars_and_points,
+                        &sr_params.even_parameters.pc_gens,
+                        &sr_params.even_parameters.bp_gens,
+                    )
+                    .unwrap();
+                    batch_verify(
+                        vesta_verification_scalars_and_points,
+                        &sr_params.odd_parameters.pc_gens,
+                        &sr_params.odd_parameters.bp_gens,
+                    )
+                    .unwrap();
                 })
             },
         );
@@ -350,7 +196,7 @@ fn bench_pour_with_parameters<
 
 criterion_group! {
     name = pour;
-    config = Criterion::default().sample_size(50);
+    config = Criterion::default().sample_size(10);
     targets =
     bench_pour,
 }
